src/jit/codegencommon.cpp

```diff
@@ -6507,6 +6507,8 @@ void CodeGen::genSetGSSecurityCookie(regNumber initReg, bool* pInitRegZeroed)
         //  mov   dword ptr [frame.GSSecurityCookie], #GlobalSecurityCookieVal
         instGen_Store_Imm_Into_Lcl(TYP_I_IMPL, EA_PTRSIZE, compiler->gsGlobalSecurityCookieVal,
                                    compiler->lvaGSSecurityCookie, 0, initReg);
+
+        *pInitRegZeroed = false;
```

Comment:
Unfortunately, this might cause x86 to zero `initReg` later when it doesn't need to, since `instGen_Store_Imm_Into_Lcl` for x86 doesn't touch `initReg`.

Reply:
Yes, you are right. And it's very hard to notice since we passing `initReg` to `instGen_Store_Imm_Into_Lcl` in x86 case (where it's going to be ignored)

Reply:
`instGen_Store_Imm_Into_Lcl` was not that useful function - 3 usages in total and its interface is misleading - having a parameter that is not used on a particular architecture. I decided to remove this function completely.